src/replica/replica_2pc.cpp

```diff
@@ -282,6 +284,7 @@ void replica::send_prepare_message(::dsn::rpc_address addr,
                                    bool pop_all_committed_mutations,
                                    int64_t learn_signature)
 {
+    mu->tracer->add_point(fmt::format("replica::send_prepare[{}]", addr.to_string()));
```

Comment:
```
tracer->add_point("p1");
statge p1
tracer->add_point("p2");
```
So statge p1's duration is represent in tracer's point p2?

Reply:
`point` means the stage `start time`, `p1's duration` means p1 start to p2 start

Reply:
For log:
```
        TRACE:name=replication_app_base::apply_mutation              , span=3439                , total=393373              , ts=1597374924883980362
        TRACE:name=complete(write2rocksdb)                           , span=107551              , total=500924              , ts=1597374924884087913

```
When I want to check how much time cost in `replication_app_base::apply_mutation`, I have to find the **NEXT** line, that is `span=107551`?

Reply:
Oh, The last point means end

Reply:
I have considered the question, named `replication_app_base::apply_mutation` let user think the tracer is tracing the `method` complete consuming, however, the tracer is trace the stage of one process, so it can actually means the process start, end and any other stage but not only focus the method complete consuming. so I rename the stage more clear to indicate the process to avoid misunderstanding it.
```
***************[TRACE:mutation[60]]***************
        TRACE:name=add request(34) to mutation                       , span=                1100, total=                1100, ts=1597819144953901791
        TRACE:name=ready to prepare                                  , span=                1886, total=                2986, ts=1597819144953903677
        TRACE:name=start prepare                                     , span=                2197, total=                5183, ts=1597819144953905874
        TRACE:name=send prepare to node[127.01.01.100:34801]         , span=                3173, total=                8356, ts=1597819144953909047
        TRACE:name=send prepare to node[127.01.01.100:34802]         , span=               46447, total=               54803, ts=1597819144953955494
        TRACE:name=shared log append mutation                        , span=               30129, total=               84932, ts=1597819144953985623
        TRACE:name=commit shared log                                 , span=                6312, total=               91244, ts=1597819144953991935
        TRACE:name=commit shared log completed                       , span=               41717, total=              132961, ts=1597819144954033652
        TRACE:name=shared log append completed                       , span=               12241, total=              145202, ts=1597819144954045893
        TRACE:name=try commit on primary                             , span=                 552, total=              145754, ts=1597819144954046445
        TRACE:name=receive prepare reply from[127.01.01.100:34801]   , span=              100883, total=              246637, ts=1597819144954147328
        TRACE:name=receive prepare reply from[127.01.01.100:34802]   , span=               56862, total=              303499, ts=1597819144954204190
        TRACE:name=try commit on primary                             , span=                3649, total=              307148, ts=1597819144954207839
        TRACE:name=ready to execute the mutation                     , span=                1041, total=              308189, ts=1597819144954208880
        TRACE:name=write mutation to rocksdb                         , span=                1075, total=              309264, ts=1597819144954209955
        TRACE:name=write mutation to rocksdb completed               , span=               89835, total=              399099, ts=1597819144954299790
        ***************[TRACE:aio_task[217298682020630414]]***************
        TRACE:name=aio task submit                                   , span=               62745, total=               62745, ts=1597819146293149672
        TRACE:name=aio task submit complete                          , span=               65348, total=              128093, ts=1597819146293215020
        TRACE:name=aio task complete                                 , span=                8562, total=              136655, ts=1597819146293223582
        TRACE:name=aio task execute callback                         , span=               15082, total=              151737, ts=1597819146293238664
```

[dsn::defer](https://github.com/apache/incubator-pegasus/blob/master/src/server/test/pegasus_server_write_test.cpp#L55) is for the method containing multi return and we need focus these end time, but now, I haven't found the case, so I don't use it.